internal/api/hcpopenshiftcluster.go

```diff
@@ -100,6 +101,36 @@ type ClusterAutoscalingProfile struct {
 	PodPriorityThreshold        int32 `json:"podPriorityThreshold,omitempty"`
 }
 
+// The ETCD settings and configuration options
+// If not specified platform managed keys are used.
+type EtcdProfile struct {
+	DataEncryption EtcdDataEncryptionProfile `json:"dataEncryption,omitempty" visibility:"read create"`
+}
+
+// EtcdDataEncryptionProfile - The ETCD data encryption settings.
+type EtcdDataEncryptionProfile struct {
+	CustomerManaged   CustomerManagedEncryptionProfile        `json:"customerManaged,omitempty" visibility:"read create"`
+	KeyManagementMode EtcdDataEncryptionKeyManagementModeType `json:"keyManagementMode,omitempty" visibility:"read create"`
+}
+
+// CustomerManagedEncryptionProfile - Customer managed encryption key profile.
+type CustomerManagedEncryptionProfile struct {
+	EncryptionType CustomerManagedEncryptionType `json:"encryptionType,omitempty" visibility:"read create"`
+	Kms            KmsEncryptionProfile          `json:"kms,omitempty" visibility:"read create"`
```

Comment:
Let's make `Kms` a pointer field and add `validate:"omitempty"`.  That may solve the unit test errors you're seeing from adding `required` to the `KmsKey` fields.  That may not be the final solution for validation but maybe good enough for this PR.  I need to give more thought to the proper validation tags.
```suggestion
	Kms            *KmsEncryptionProfile          `json:"kms,omitempty" visibility:"read create" validate:"omitempty"`
```